model/src/main/kotlin/config/PackageConfiguration.kt

```diff
@@ -66,25 +68,55 @@ data class PackageConfiguration(
      * License finding curations.
      */
     @JsonInclude(JsonInclude.Include.NON_EMPTY)
-    val licenseFindingCurations: List<LicenseFindingCuration> = emptyList()
+    val licenseFindingCurations: List<LicenseFindingCuration> = emptyList(),
+
+    /**
+     * The source code origin this configuration applies to.
+     */
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    val sourceCodeOrigin: SourceCodeOrigin? = null
```

Comment:
Would it be possible to split this commit up into 2, to simplify the review?

1. Add source code origin
2. Allow version ranges

Not sure about the ordering.

Reply:
In this first PR `Add source code origin`, would you expect also some minimal usage of `sourceCodeOrigin` in `matches`, or could if be done in following commit ?

Reply:
> would you expect also some minimal usage of `sourceCodeOrigin`

yes.

Reply:
Opened https://github.com/oss-review-toolkit/ort/pull/10377.